### ISSUE TITLE
Support LAST_INSERT_ID simple pass through

### ISSUE THREAD

**Issue by tirsen** [youtube/vitess#2811]
Support LAST_INSERT_ID for unsharded tables or when targetting a specific shard by simply passing through the invocation unchanged. Note we need support for both the nullary and unary versions.

Specifically I need support for the somewhat idiomatic `LAST_INSERT_ID(LAST_INSERT_ID()+1)` used in a multi-row `UPDATE` statement. (Used to assign a new ID for each row.)

**Comment by tirsen** [comment]
Actually in our case we need to support what's mentioned in the documentation here:
https://dev.mysql.com/doc/refman/5.7/en/information-functions.html#function_last-insert-id

I.e. you first call this to reset the "implicit" sequence to the base value
```
SELECT LAST_INSERT_ID(?)
```

Then you call this to set set a sequence of numbers in a multi-row update statement
```
UPDATE ... 
SET some_id = LAST_INSERT_ID(LAST_INSERT_ID()+1) 
WHERE ...
```

**Comment by bbeaudreault** [comment]
We also need `LAST_INSERT_ID`. Our primary use case is for upsert on unsharded keyspaces. It can be used in the update portion to properly handle auto increment. 

For example: http://stackoverflow.com/a/779252

**Comment by bbeaudreault** [comment]
@sougou thoughts on this now that the other v3 stuff is mostly wrapped up?

**Comment by bbeaudreault** [comment]
I think it's also in line with the remaining task on the other issue on supporting unsharded keyspaces without `keyspace/0`

**Comment by sougou** [comment]
Yeah. I'm refactoring the v3 code to accommodate more constructs.

**Comment by sougou** [comment]
TL;DR: I'll be adding support for `LAST_INSERT_ID` as pass-through for unsharded keyspaces only. The caveat is that the value will be usable only within the scope of a transaction. If you start a new transaction, you will get a new connection for it and the previous value that you set will not be reusable.

We could eventually look at supporting `last_insert_id` as a session variable in Vitess. Work for this is medium complexity, and we can make it work for sharded databases also. We can also integrate the behavior with vitess sequences.

However, the use case linked here http://stackoverflow.com/a/779252 by @bbeaudreault is a little harder to support for a sharded keyspace because it combines auto-inc, upsert, last_insert_id, and on a table with multiple unique keys. So, for now, we'll limit ourselves to sending it as pass-through for unsharded keyspaces only.

**Comment by tirsen** [comment]
Sweet! Now, can I get it for sharded tables too? (backed by autoinc sequences obviously)
